deliver/lib/deliver/app_trailer.rb

```diff
@@ -0,0 +1,60 @@
+module Deliver
+  # AppTrailer represents one video trailer for one specific locale and
+  # device type.
+  class AppTrailer
+
+    attr_accessor :device_type
+
+    attr_accessor :video_path
+
+    attr_accessor :language
+
+    attr_accessor :timestamp
+
+    attr_accessor :preview_image_path
+
+    # @param video_path (String) path to the video file
+    # @param language (String) Language of this trailer (e.g. en-US)
+    def initialize(video_path, language)
+      self.video_path = video_path
+      self.language = language
+
+      # should be given as video file name prefix since we can't determine device by resolution
+      self.device_type = self.class.discover_device_type(video_path)
+      self.timestamp = '05.00'
+      # should be in the same directory as video file and also prefixed with device type
+      self.preview_image_path = self.class.discover_preview_image(video_path, device_type)
+
+      UI.user_error!("The extension of given video (#{video_path}) does not match the requirements (mov, m4v, mp4)") unless self.valid_file_extension?
```

Comment:
maybe move that line inside a `validate_file_extension!` to have both list of valid extensions in one place.